### PR TITLE
feat added stub to modules gatk4_mergebamalignment_mutect2_revertsam_samtofastq, samtools_view

### PR DESCRIPTION
**Pull request by Lucpen**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by ramprasadn** [comment]
@Lucpen Nice work! Don't forget to add tests for stubs like [here](https://github.com/nf-core/modules/pull/1543/files#diff-63489f078a8346967ccbd4634a18c09120dcf04fe56b9ecb1e679038286c6af2=) 😄

File: tests/modules/gatk4/mergebamalignment/main.nf [hunk 1]
```diff
@@ -14,3 +14,14 @@ workflow test_gatk4_mergebamalignment {
...
+
+workflow test_gatk4_mergebamalignment_stubs {
+    input    = [ [ id:'test' ], // meta map
+                 file(params.test_data['sarscov2']['illumina']['test_single_end_bam'], checkIfExists: true),
+                 file(params.test_data['sarscov2']['illumina']['test_unaligned_bam'], checkIfExists: true)
+               ]
+    fasta    = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+    dict     = file(params.test_data['sarscov2']['genome']['genome_dict'], checkIfExists: true)
+
+    GATK4_MERGEBAMALIGNMENT ( input, fasta, dict )
```
Comment (maxulysse) [inline tests/modules/gatk4/mergebamalignment/main.nf:26]: If we tests stubs, do we need an actual real file?
Could this be working?
```suggestion
    input    = [ [ id:'test' ], // meta map
                 "test_foo.bam",
                 "test_bar.bam"
               ]
    fasta    = "genome.fasta"
    dict     = "genome.fasta.dict"

    GATK4_MERGEBAMALIGNMENT ( input, fasta, dict )
```

Comment (Lucpen) [reply tests/modules/gatk4/mergebamalignment/main.nf]: Good point! I have updated it on all the modules changed.

Comment (ramprasadn) [reply tests/modules/gatk4/mergebamalignment/main.nf]: If this works, do you think this should be the standard for testing stubs? ;) The advantage I see for not using actual files is that they wouldn't have to be staged in the work folder, so we get a leaner work environment.

File: tests/modules/gatk4/mutect2/main.nf [hunk 2]
```diff
@@ -118,3 +118,25 @@ workflow test_gatk4_mutect2_mitochondria {
...
     GATK4_MUTECT2_MITO ( input, fasta, fai, dict, [], [], [], [] )
 }
+
+workflow test_gatk4_mutect2_tumor_normal_pair_f1r2_stubs {
+    input = [ [ id:'test', normal_id:'normal', tumor_id:'tumour' ], // meta map
+              [ "foo_parired.bam",
+                "foo_parired2.bam"
+                ],
+              [ "foo_parired.bam.bai",
+                "foo_parired2.bam.bai"
```
Comment (maxulysse) [inline tests/modules/gatk4/mutect2/main.nf:128]:
```suggestion
              [ "foo_paired.bam",
                "foo_paired2.bam"
                ],
              [ "foo_paired.bam.bai",
                "foo_paired2.bam.bai"
```

File: tests/modules/gatk4/mergebamalignment/main.nf (hunk 1 above)
Comment (maxulysse) [reply tests/modules/gatk4/mergebamalignment/main.nf]: Definitively, never really thought of that before actually seeing that here

Comment (Lucpen) [reply tests/modules/gatk4/mergebamalignment/main.nf]: It seems to work ^^

**Review by ramprasadn — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**